internal/pkg/cli/env_init.go

```diff
@@ -565,6 +596,20 @@ func (o *initEnvOpts) validateCredentials() error {
 	return nil
 }
 
+func uploadLambdaToS3(reader reader, uploader zipAndUploader, bucket string) error {
```

Comment:
What do you think of actually moving this logic to the `template` pkg instead? I feel like we're leaking information to the `cli` pkg with:
```go
var envLambdas = map[string]string{
	"dns-cert-validator": "dns-cert-validator.js",
	"dns-delegation":     "dns-delegation.js",
	"enable-long-arns":   "enable-long-arns.js",
}
```
Instead maybe we can have a method like this:
```go
// Now template.Content satisfies the s3.NamedBinary interface.
func (c *Content) Name() string { ... }

func (t *Template) UploadCustomResources(func upload([]s3.NamedBinary) error) {
    // Gather all custom resources in a []*Content slice.
    return upload(contents...)
}
```
and finally in the `cli` package we can do:
```go
t.UploadCustomResources(func(files []s3.NamedBinary) error {
   return o.newS3(sess).ZipAndUpload(bucket, key, files)
})
```

Reply:
gotcha. addressed